sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/Metrics/ExecutableLinesMetric.cs

```diff
@@ -0,0 +1,182 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.VisualBasic;
+using Microsoft.CodeAnalysis.VisualBasic.Syntax;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Metrics.VisualBasic
+{
+    public static class ExecutableLinesMetric
+    {
+        public static ImmutableArray<int> GetLineNumbers(SyntaxTree syntaxTree, SemanticModel semanticModel)
+        {
+            var walker = new ExecutableLinesWalker(semanticModel);
+            walker.Visit(syntaxTree.GetRoot());
+
+            return walker.ExecutableLines.ToImmutableArray();
+        }
+
+        private class ExecutableLinesWalker : VisualBasicSyntaxWalker
+        {
+            private readonly HashSet<int> executableLineNumbers = new HashSet<int>();
+            private readonly SemanticModel semanticModel;
+
+            public ExecutableLinesWalker(SemanticModel semanticModel)
+            {
+                this.semanticModel = semanticModel;
+            }
+
+            public ICollection<int> ExecutableLines => this.executableLineNumbers;
+
+            public override void DefaultVisit(SyntaxNode node)
+            {
+                if (FindExecutableLines(node))
+                {
+                    base.DefaultVisit(node);
+                }
+            }
+
+            private bool FindExecutableLines(SyntaxNode node)
+            {
+                switch (node.Kind())
+                {
+                    // The following C# constructs have no equivalent in VB.NET:
+                    // - checked
+                    // - unchecked
+                    // - fixed
+                    // - unsafe
+                    // - yield break
+                    // - yield return
```

Comment:
`yield return something` should be just `Yield something` https://docs.microsoft.com/en-us/dotnet/visual-basic/language-reference/statements/yield-statement

`yield break` is `Exit Function` or `Return` (same article)... I am not sure if using `Exit Function` is a good idea in general, but I guess we should handle it as executable.